Meziantou.FluentAssertionsAnalyzers.Tests/NUnitToFluentAssertionsAnalyzerUnitTests.cs

```diff
@@ -51,7 +51,69 @@ public void MyTest()
 }
 """);
     }
-    
+
+    [Fact]
+    public Task Assert_Dynamic()
+    {
+        return Assert(
+"""
+using NUnit.Framework;
+
+class Test
+{
+    public void MyTest()
+    {
+        dynamic val = new object();
+        [|Assert.That(val, Is.Not.Null)|];
+    }
+}
+""",
+"""
+using FluentAssertions;
+using NUnit.Framework;
+
+class Test
+{
+    public void MyTest()
+    {
+        dynamic val = new object();
+        val.Should().NotBeNull();

```

Comment:
Are you sure this syntax is valid?

````
dynamic a = new object();
a.Should().BeNull();
````

> Unhandled exception. Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'object' does not contain a definition for 'Should'

Reply:
Yes, that's a very fair point... 🤦 

As per https://stackoverflow.com/a/71676317/532575, I'm casting it to an `object` if needed.

It feels a bit ugly, and the resulting code is slightly more complicated too, so I won't be offended if you don't want to merge it at all. ;)